sessionctx/variable/sysvar.go

```diff
@@ -856,6 +856,16 @@ var defaultSysVars = []*SysVar{
 		s.EnableAlterPlacement = TiDBOptOn(val)
 		return nil
 	}},
+	{Scope: ScopeGlobal | ScopeSession, Name: TiDBPlacementMode, Value: DefTiDBPlacementMode, Type: TypeStr, Validation: func(vars *SessionVars, normalizedValue string, originalValue string, scope ScopeFlag) (string, error) {
+		mode := PlacementMode(normalizedValue)
+		if !mode.Valid() {
+			return normalizedValue, ErrWrongTypeForVar.GenWithStackByArgs(TiDBPlacementMode)
+		}
+		return string(mode), nil
```

Comment:
You can actually remove the validation func completely here - just change the type to `TypeEnum` with the permitted values for strict/ignore.

Reply:
Sorry, I'm not very understand. Could you explain it about change the type to `TypeEnum` ?

Reply:
See https://github.com/pingcap/tidb/blob/fb03dacbcaa11145003838d672a6956ac60d7ff6/sessionctx/variable/sysvar.go#L606-L609 for an example.

The `TypeEnum` validates that options belong to the list of `PossibleValues`, and will return an error if they are invalid. It is also case insensitive and normalizes to the case in the `PossibleValues`.

(We have many such cases for enumerated options, which is why using the built-in helper is useful).

Reply:
Okay. Thank you